test/e2e/specs/cart.overview.spec.js

```diff
@@ -0,0 +1,74 @@
+import AppHeader from '../page-objects/AppHeader';
+import SwagOverview from '../page-objects/SwagOverview';
+import CartOverview from '../page-objects/CartOverview';
+import CheckoutPersonalInfo from '../page-objects/CheckoutPersonalInfo';
+import {prepareEnvironment} from '../helpers';
+
+describe('Cart', () => {
+    it('should validate that we can continue shopping', () => {
+        prepareEnvironment('/cart.html');
+
+        CartOverview.waitForIsDisplayed();
```

Comment:
I usually do something like:
```
CartPage.open();
expect(CartPage.isVisible()).toEqual(true); //the synchronization would happen in here rather than a direct call to `waitForIsDisplayed()` 
```
The benefit here is that if this assertion fails, we do not move on. Basically seems the same as your approach but one less line of code in the tests. Open page, Assert it's loaded, Proceed with actions. As opposed to Open Page, Wait, Assert is loaded, Proceed with actions. Thoughts?

Reply:
Hmm,

I don't really understand what you mean, the `CartOverview.waitForIsDisplayed()`fails when the page is not loaded and will not go to the next page, so I thought an extra assertion would not be useful, secondly it's not the scope of the test

Reply:
No big deal, I'm happy to move past this one

Reply:
I prefer Wim's approach, explicit without an assertion.